src/borg/remote.py

```diff
@@ -1141,31 +1159,56 @@ def get_many(self, keys, cache=True):
                 file = self.key_filename(key)
                 with open(file, 'rb') as fd:
                     self.hits += 1
-                    yield fd.read()
+                    yield self.unpack(fd.read())
             else:
                 for key_, data in repository_iterator:
                     if key_ == key:
-                        if cache:
-                            self.add_entry(key, data)
+                        transformed = self.add_entry(key, data, cache)
                         self.misses += 1
-                        yield data
+                        yield transformed
                         break
                 else:
                     # slow path: eviction during this get_many removed this key from the cache
                     t0 = time.perf_counter()
                     data = self.repository.get(key)
                     self.slow_lat += time.perf_counter() - t0
-                    if cache:
-                        self.add_entry(key, data)
+                    transformed = self.add_entry(key, data, cache)
                     self.slow_misses += 1
-                    yield data
+                    yield transformed
         # Consume any pending requests
         for _ in repository_iterator:
             pass
 
 
-def cache_if_remote(repository):
+def cache_if_remote(repository, *, decrypted_cache=False, pack=None, unpack=None, transform=None):
+    """
+    Return a Repository(No)Cache for *repository*.
+
+    If *decrypted_cache* is a key object, then get and get_many will return a tuple
+    (csize, plaintext) instead of the actual data in the repository. The cache will
+    store decrypted data, which increases CPU efficiency (by avoiding repeatedly decrypting
+    and more importantly MAC and ID checking cached objects).
```

Comment:
= b544af2. I removed the automatic recovery part in 4faaa7d though, since these would _usually_ ($TMP could be on a disk, but its rare nowadays) indicate bad RAM. I leave the changesets in though, so if we decide later to have that we can.